### DISCUSSION TITLE
File name concatenation with public path in asset modules.

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#13593] [Q&A] [answered]
I have asset modules setup.

My configuration is 

```js
{
 test: /\.svg$/i,
 type: 'asset/resource',
 issuer: {
    and: [/\.jsx$/]
  },
  generator: {
    filename: 'assets/[name].[ext]',
  }
}
```

and the publicPath is `http://localhost:3000`.

the generated code has asset path `http://localhost:3000assets/[name].[ext]`. Is this the expected behaviour and mandatory to include trailing slash at the end of the URL's? or a bug?. The same public path seems to be working fine with other assets like javascript without trailing slash.

**Comment by alexander-akait** [comment, marked as answer]
Please create reproducible test repo

**Reply by ghost** [reply]
Sure, thanks. 

Here. 

[public-path-issue.zip](https://github.com/webpack/webpack/files/6677393/public-path-issue.zip).

I created bare minimal code.

Steps:
- npm run serve -  opens the port at 8080. 
- Open `http://localhost:8080` and check the console to see the path of the svg

**Reply by alexander-akait** [reply]
You have two problems here:
- You need slash at the end of the `publicPath` option (we describe it here https://webpack.js.org/configuration/output/#outputpublicpath):
```
output: {
    publicPath: 'http://localhost:8080/',
  },
```
- Don't use `.` for `filename`, i.e. please use:
```
generator: {
  filename: 'assets/[name][ext]',
}
```

**Reply by ghost** [reply]
Thanks @alexander-akait . Now I see it. 

> Don't use . for filename, i.e. please use:

Thanks. I was switching from file-loader. I missed it
